nservicebus/azure-service-bus/message-lock-renewal.md

```diff
@@ -0,0 +1,28 @@
+---
+title: Azure Service Bus Transport Message lock renewal
+summary: Extending message lock renewal
+component: ASB
+tags:
+- Cloud
+- Azure
+- Transports
+reviewed: 2016-04-20
+---
+
+When messages are received using `PeekLock` mode, the receive operation becomes a two-stage operation. The first stage, the message is locked by the broker for a specific consumer for a fixed period known as `LockDuration`. When the consumer is done with the message, the message is marked as completed by the consumer, indicating to the broker that second stage is finished. Default lock duration for the Azure Service Bus transport is set to 30 seconds. Maximum lock duration allowed by the broker service is 5 minutes. `LockDuration` is a global setting and its value applied to all entities.
+
+Occasionally, processing can take longer than the maximum allowed time for `LockDuration`. As a result of that, messages will re-appear on the queue and will be available to other consumers. Azure Service Bus transport version 7 and above supports automatic message lock renewal. 
+
+
+## How does message lock renewal work?
+
+Message lock renewal should be greater than `LockDuration`. Whenever `LockDuration` period is about to expire, Azure Service Bus transport will issue lock renewal to the broker, keeping message locked and invisible to other consumers. Lock renewal will automatically take place while the total time of the message processing stays less than auto renewal time set by Azure Service Bus transport. The default lock renewal time is 5 minutes. Auto lock renewal will **not** increase `DeliveryCount` of the processed message.
+
+
+## Configuring message lock renewal
+
+The configuration of message lock renewal is done on `MessageReceivers` extension of the transport, specifying the maximum period lock renewal should take place.
+
+snippet: asb-auto-lock-renewal
+
+For example, setting lock renewal time to 10 minutes using `LockDuration` set to 1 minute will ensure that message will remain locked up to 10 minutes, but not past that. If message processing exceeds the 10 minutes, the message will become available to other consumers.
```

Comment:
No, message becomes available for other consumers to be consumed. No rollbacks (no transactions). Whenever the original consumer will attempt to complete the message, it will be rejected since it's no longer has a valid lock over that message.